src/utils/parserUtils.ts

```diff
@@ -34,30 +33,41 @@ export function parse(code: string): any {
 };
 
 /**
- * Finds all nodes which have one of several types in a syntax tree.
+ * Finds all nodes which a type in a syntax tree.
  * @function
  * @param abstractSyntaxTree - The current part of the abstract syntax tree which will be investigated.
- * @param  types - The list of types which are requested.
- * @returns  All nodes which have one of the requested types.
+ * @returns  All nodes with a type.
  */
-export function getNodesWithType(abstractSyntaxTree: any, types: string[], nodes?: any[], key?: string): any[] {
```

Comment:
This method name is not descriptive. What is it doing? Is the key-parameter ever used? Maybe it should be called `collectNodes` or something, or `collectAndFreezeNodes` (although i'm not a fan of `and` in method names). What do you think?